packages/revocation/src/implementations/revocationOnChain.ts

```diff
@@ -0,0 +1,127 @@
+import {
+  Contract, ethers, Event, utils,
+} from 'ethers';
+import { EwSigner, addressOf } from '@ew-did-registry/did-ethr-resolver';
+import { abi as RevocationOnChainAbi } from '../constants/RevocationRegistryOnChain.json';
+
+export class RevocationOnChain {
+  private _revocationRegistryOnChain: Contract;
+
+  /**
+  * @param owner - Entity which controls revocation
+  * @param addressOnchain - Address of the on chain claim's RevocationRegistry
+  */
+  constructor(
+    owner: EwSigner,
+    addressOnChain: string,
+  ) {
+    this._revocationRegistryOnChain = new ethers.Contract(
+      addressOnChain,
+      RevocationOnChainAbi,
+      owner,
+    );
+  }
+
+  /**
+  * Revokes the on chain role
+  * Returns true on success
+  *
+  * @param { string } role - role to be revoked
+  * @param { string } subject - DID of which the role is being revoked
+  * @param { string } revoker - DID of the revoker
+  * @returns Promise<boolean>
+  */
+  async revokeRole(role: string, subject: string, revoker: string):
+  Promise<boolean> {
+    const roleHash = utils.namehash(role);
+    const subjectAddress = addressOf(subject);
+    const revokerAddress = addressOf(revoker);
+    try {
+      const tx = await this._revocationRegistryOnChain.revokeClaim(
+        roleHash,
+        subjectAddress,
+        revokerAddress,
+      );
+      const receipt = await tx.wait();
+      const event = receipt.events.find(
+        (e: Event) => (e.event === 'Revoked'),
+      );
+      if (!event) return false;
+    } catch (error) {
+      throw new Error(error);
+    }
+    return true;
+  }
+
+  /**
+  * Revokes the role for list of DIDs
+  * Returns true on success
+  *
+  * @param { string } role - role to be revoked for the DIDs
+  * @param { string[] } subjects - DIDs for which the role is being revoked
+  * @param { string } revoker - DID of the revoker
+  * @returns Promise<boolean>
+  */
+  async revokeRoleforDIDs(role: string, subjects: string[], revoker: string):
+  Promise<boolean> {
+    const roleHash = utils.namehash(role);
+    const revokerAddress = addressOf(revoker);
+    const revocationSubjects = [];
+    for (let i = 0; i < subjects.length; i++) {
```

Comment:
```suggestion
    for await (let i = 0; i < subjects.length; i++) {
```

Reply:
This throws error and I guess it's not needed as the contract call already has `await`.